classes/Customer.php

```diff
@@ -1184,10 +1179,12 @@ public function transformToCustomer($idLang, $password = null)
         if (!$this->isGuest()) {
             return false;
         }
+
         if (empty($password)) {
-            $password = Tools::passwdGen(8, 'RANDOM');
+            $password = Tools::passwdGen(16, 'RANDOM');
         }
-        if (!Validate::isPlaintextPassword($password)) {
+
+        if (!Validate::isAcceptablePasswordLength($password) or !Validate::isAcceptablePasswordScore($password)) {
```

Comment:
Not used to the `or` in the rest of the code base, maybe use `||`?
```suggestion
        if (!Validate::isAcceptablePasswordLength($password) || !Validate::isAcceptablePasswordScore($password)) {
```